pkg/arvo/sys/vane/eyre.hoon

```diff
@@ -997,11 +997,128 @@
+        "Invalid input: Expected /~/boot/<ship=@p>"
+      ==
+    =/  ship
+      %+  slaw
+        %p
+      i.t.t.crumbs
+    ?~  ship
+      %^  return-static-data-on-duct  400  'text/html'
+      %:  error-page
+        400
+        &
+        url.request
+        "Invalid input: Expected /~/boot/<ship=@p>"
+      ==
+    %^  return-static-data-on-duct  200  'text/plain'
+    (as-octs:mimes:html (scot %p (galaxy-for u.ship)))
+  ::  Takes two path parameters - ship=@p and optional bone=@u.
+  ::
+  ::  If only ship is provided, responds with the @p of the current sponsor,
+  ::  life, and rift of that ship, or errors with 404 status when the provided
+  ::  ship is not a known peer.
+  ::
+  ::  If both ship and bone are provided, responds with the last acked msg
+  ::  sequence number of the provided ship and bone, or errors with 404 if either
+  ::  peer or bone is not found
+  ::
+  ++  handle-boot
+    |=  [=identity =request:http]
+    ^-  (quip move server-state)
+    ?.  =(%'GET' method.request)
+      %^  return-static-data-on-duct  405  'text/html'
+      (error-page 405 & url.request "may only GET boot data")
+    =/  crumbs  q:(rash url.request apat:de-purl:html)
+    ?.  ?=([@t @t @t *] crumbs)
+      %^  return-static-data-on-duct  400  'text/html'
+      %:  error-page
+        400
+        &
+        url.request
+        "Invalid input: Expected /~/boot/<ship=@p> or /~/boot/<ship=@p>/<bone=@u>"
+      ==
+    =/  ship
+      %+  slaw
+        %p
+      i.t.t.crumbs
+    =/  bone
+      %+  slaw
+        %ud
+      ?:  ?=([@t @t @t @t ~] crumbs)
+        i.t.t.t.crumbs
+      ''
+    ?:  ?|  ?=(~ ship)
+            ?&(?=([@t @t @t @t ~] crumbs) ?=(~ bone))
+        ==
+      %^  return-static-data-on-duct  400  'text/html'
+      %:  error-page
+        400
+        &
+        url.request
+        "Invalid input: Expected /~/boot/<ship=@p> or /~/boot/<ship=@p>/<bone=@u>"
+      ==
+    =/  des=(unit (unit cage))
+      %:  rof
+        [~ ~]
+        /eyre
+        %ax
+        [our %$ da+now]
+        /peers/(scot %p u.ship)
+      ==
+    ?.  ?=([~ ~ %noun *] des)
+      %^  return-static-data-on-duct  404  'text/html'
+      (error-page 404 & url.request "Peer {(scow %p u.ship)} not found.")
+    =/  ship-state
+      !<  ship-state:ames  q.u.u.des
+    ?.  ?=(%known -.ship-state)
+      !!
+    ?~  bone
+      %^  return-static-data-on-duct  200  'text/plain'
+      (as-octs:mimes:html (scot %u (jam [(galaxy-for u.ship) rift.-.+.ship-state life.-.+.ship-state])))
+    =/  rcv=(map bone:ames message-sink-state:ames)  rcv.+.ship-state
+    =/  mss=(unit message-sink-state:ames)  (~(get by rcv) u.bone)
+    ?~  mss
+      %^  return-static-data-on-duct  404  'text/html'
+      %:  error-page
+        404
+        &
+        url.request
+        "Bone {(scow %u u.bone)} of peer {(scow %p u.ship)} not found."
+      ==
+    %^  return-static-data-on-duct  200  'text/plain'
+    (as-octs:mimes:html (scot %u last-acked.u.mss))
```

Comment:
This should return the rift as well, the aura should be `%uw`, the type should be named and versioned in lull.